charms/istio-pilot/src/charm.py

```diff
@@ -352,35 +353,62 @@ def _istiod_svc(self):
         else:
             return exporter_ip
 
-    @property
-    def _gateway_address(self):
-        """Look up the load balancer address for the ingress gateway.
-        If the gateway isn't available or doesn't have a load balancer address yet,
-        returns None.
-        """
-        # FIXME: service name is hardcoded
+    def _is_gateway_service_up(self):
+        """Returns True if the gateway is up, else False."""
+        svc = self._get_gateway_service()
+
+        if svc.spec.type == "NodePort":
+            # TODO: do we need to interrogate this further for status?
```

Comment:
Maybe? Is there some `status` or `state` field in a `NodePort`?

Reply:
Not that I could see.  It is not a very descriptive object :|

Reply:
Yeh, I don't see how we could do it, but let's keep an eye on it.